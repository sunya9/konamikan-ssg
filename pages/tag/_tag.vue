<template>
  <div>
    <app-header
      home-button
      :title="tag.name"
      :description="tag.description"
      :cover="$resolveUrl(tag.feature_image)"
    />
    <main class="section">
      <div class="container">
        <posts-grouped-by-year-wrapper :posts="posts" />
      </div>
    </main>
  </div>
</template>
<script lang="ts">
import Vue from 'vue'
import { Component } from 'nuxt-property-decorator'
import { PostObject, Tag, TagsObject } from '@tryghost/content-api'
import PostsGroupedByYearWrapper from '~/components/PostsGroupedByYearWrapper.vue'
import AppHeader from '~/components/AppHeader.vue'
import { reducePostFieldMapper, PostOrPageLight } from '~/util/util'

@Component({
  components: {
    PostsGroupedByYearWrapper,
    AppHeader
  },
  async asyncData({ $http, route, error }) {
    const [postsWrapper, tagsWrapper] = await Promise.all([
      $http.$get<PostObject>('/resources/posts'),
      $http.$get<TagsObject>('/resources/tags')
    ])
    const posts = postsWrapper.posts
    const tags = tagsWrapper.tags
    const tag = tags.find((tag) => tag.slug === route.params.tag)
    if (!tag) return error({ statusCode: 404 })
    return {
      posts: posts
        .filter((post) =>
          post.tags?.some((postTag) => postTag.slug === tag.slug)
        )
        .map(reducePostFieldMapper),
      tag
    }
  }
})
export default class TagPage extends Vue {
  tag!: Tag
  posts!: PostOrPageLight[]
  head() {
    const meta: object[] = [
      {
        hid: 'description',
        property: 'description',
        content: this.tag.description
      },
      ...this.$createBaseMetadata(this.tag)
    ]
    return {
      title: this.tag.meta_title || this.tag.name,
      meta
    }
  }
}
</script>
