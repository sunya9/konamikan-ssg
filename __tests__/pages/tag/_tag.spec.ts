import { mount } from '@vue/test-utils'
import { post, postWithFeatureImage } from '@/__tests__/fixtures/postOrPage'
import Tag from '@/pages/tag/_tag.vue'
import { tag } from '~/__tests__/fixtures/tag'

describe('tag page', () => {
  it('normal', () => {
    expect(
      mount(Tag, {
        mocks: {
          posts: [post, postWithFeatureImage],
          tag
        }
      })
    ).toMatchSnapshot()
  })
})
