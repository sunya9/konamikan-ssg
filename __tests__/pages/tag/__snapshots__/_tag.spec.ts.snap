// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tag page normal 1`] = `
<div>
  <header class="hero is-medium fix-background" style="background-image: url(cover.png);">
    <div class="hero-head">
      <nav class="navbar ">
        <div class="container">
          <div class="navbar-brand">
            <div class="navbar-brand"><a class="navbar-item has-text-white-bis">
                title
              </a></div> <a href="#" aria-label="Menu" class="navbar-burger has-text-white"><span aria-hidden="true"></span> <span aria-hidden="true"></span> <span aria-hidden="true"></span></a>
          </div>
          <div class="navbar-menu is-relative">
            <div class="navbar-end">
              <div class="navbar-item">
                <div class="control has-icons-right"><label for="search-form-input" class="is-sr-only">Search</label> <input id="search-form-input" type="search" placeholder="Search..." autocomplete="off" class="input is-small search"> <span class="icon is-small is-right"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-search"><circle cx="11" cy="11" r="8"></circle><line x1="21" y1="21" x2="16.65" y2="16.65"></line></svg></span></div>
                <!---->
              </div> <a href="/rss" class="navbar-item has-text-white-ter"><span class="icon"><svg xmlns="http://www.w3.org/2000/svg" width="24px" height="24px" viewBox="0 0 24 24" fill="none" stroke="currentColor" stroke-width="2" stroke-linecap="round" stroke-linejoin="round" class="feather feather-rss"><path d="M4 11a9 9 0 0 1 9 9"></path><path d="M4 4a16 16 0 0 1 16 16"></path><circle cx="5" cy="19" r="1"></circle></svg></span> <span class="is-sr-only">RSS</span></a>
            </div>
          </div>
        </div>
      </nav>
    </div>
    <div class="hero-body fix-z-index">
      <div class="container has-text-centered">
        <h1 class="title is-1 has-text-white">
          tag
        </h1>
        <!---->
      </div>
    </div>
    <!---->
  </header>
  <main class="section">
    <div class="container">
      <div>
        <div class="content">
          <h2 id="#2020"><span>2020</span> <small class="has-text-grey is-size-6">
              2 posts
            </small></h2>
          <ul>
            <li>
              <div class="oneline"><a><span class="has-text-grey">
          04/05
        </span> <span>title</span> <span class="has-text-grey excerpt">
          excerpt
        </span></a></div>
            </li>
            <li>
              <div class="oneline"><a><span class="has-text-grey">
          04/05
        </span> <span>title</span> <span class="has-text-grey excerpt">
          excerpt
        </span></a></div>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </main>
</div>
`;
